src/algebra/star/basic.lean

```diff
@@ -118,32 +120,45 @@ local attribute [instance] star_monoid_of_comm
 end
 
 /--
-A `*`-ring `R` is a (semi)ring with an involutive `star` operation which is additive
-which makes `R` with its multiplicative structure into a `*`-monoid
-(i.e. `star (r * s) = star s * star r`).
+A `*`-additive monoid `R` is an additive monoid with an involutive `star` operation which
+preserves addition.
 -/
-class star_ring (R : Type u) [semiring R] extends star_monoid R :=
+class star_add_monoid (R : Type u) [has_add R] extends has_involutive_star R :=
```

Comment:
It would be more consistent with what's done on the other algebraic classes to use `[add_monoid R]` rather than `[has_add R]`.

Reply:
Which other algebraic classes are you thinking of?

Reply:
For example,
```lean
class star_monoid (R : Type u) [monoid R] extends has_involutive_star R :=
class star_ring (R : Type u) [semiring R] extends star_monoid R :=
```